kpet/data.py

```diff
@@ -329,6 +379,64 @@ def is_dir_valid(dir_path):
         """
         return os.path.isfile(dir_path + "/index.yaml")
 
+    def validate(self):
+        """
+        Check if any of the regexes are invalid.
+
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+        host_types = list()
+        trees = list()
+        arches = list()
+
+        for host_name in self.host_types or []:
+            host_types.append(host_name)
```

Comment:
Can't you write `host_types = list((self.host_types or {}).keys())` instead, or something like that? After that we won't need the three initialization lines at the start.